lib/pavilion/scheduler_plugins.py

```diff
@@ -43,10 +75,98 @@ def remove_scheduler_plugin( scheduler_plugin ):
         del _SCHEDULER_PLUGINS[ name ]
 
 def get_scheduler_plugin( name ):
-    global _SCHEDULER_PLUGINS
-    if name not in _SCHEDULER_PLUGINS:
-        raise PluginSchedulerError("Module not found: '{}'".format(name))
+    global _LOADED_PLUGINS
+
+    if _LOADED_PLUGINS is None:
+        raise SchedulerPluginError(
+                              "Trying to get plugins before they are loaded." )
+
+    if name not in _LOADED_PLUGINS:
+        raise SchedulerPluginError("Module not found: '{}'".format(name))
+
+    return _LOADED_PLUGINS[ name ]
+
+class SchedulerPlugin(IPlugin.IPlugin):
+
+    PRIO_DEFAULT = 0
+    PRIO_COMMON = 10
+    PRIO_USER = 20
+
+    NAME_VERS_RE = re.compile(r'^[a-zA-Z0-9_.-]+$')
+
+    def __init__( self, name, priority=PRIO_DEFAULT ):
+        """Scheduler plugin that is expected to be overriden by subclasses.
+        The plugin will populate a set of expected 'sched' variables."""
+
+        super().__init__()
+
+        self.logger = logging.getLogger( 'sched.' + name )
+        self.name = name
+        self.priority = priority
+        self.values = {}
+        for var in _SCHEDULER_VARS:
+            self.values[ var ] = None
+
+    def _get( var ):
+        raise NotImplemented
+
```

Comment:
Oh, I see. You're treating the scheduler similar to the sys variables. I'm pretty sure that's not going to cut it. 

We need to have a sit down design session on this.